src/wayland/dmabuf/dispatch.rs

```diff
@@ -330,28 +310,18 @@ where
                 // create_dmabuf performs an implicit ensure_unused function call.
                 if let Some(dmabuf) = data.create_dmabuf(params, width, height, format, flags) {
                     if state.dmabuf_state().globals.get(&data.id).is_some() {
-                        match state.dmabuf_imported(&DmabufGlobal { id: data.id }, dmabuf.clone()) {
-                            Ok(_) => {
-                                // Import was successful, initialize the dmabuf data
-                                data_init.init(buffer_id, dmabuf);
-                            }
-
-                            Err(ImportError::InvalidFormat) => {
-                                params.post_error(
-                                    zwp_linux_buffer_params_v1::Error::InvalidFormat,
-                                    "format and plane combination are not valid",
-                                );
-                            }
-
-                            Err(ImportError::Failed) => {
-                                // Buffer import failed. The protocol documentation heavily implies killing the
-                                // client is the right thing to do here.
-                                params.post_error(
-                                    zwp_linux_buffer_params_v1::Error::InvalidWlBuffer,
-                                    "buffer import failed",
-                                );
-                            }
-                        }
+                        // The buffer isn't technically valid during data_init, but the client is not allowed to use the buffer until ready.
+                        let buffer = data_init.init(buffer_id, dmabuf.clone());
+
+                        let notifier = ImportNotifier {
+                            inner: params.clone(),
+                            display: dh.clone(),
+                            dmabuf: dmabuf.clone(),
+                            import: Import::Infallible(buffer),
+                            drop_ignore: false,
```

Comment:
Could we have a constructor for this?

Reply:
I assume you mean an internal constructor?

Reply:
Yes. `drop_ignore` for example is always `false` and doesn't need to be part of it.